test-common/test-common-api/src/main/java/org/apache/kafka/common/test/api/ClusterInstance.java

```diff
@@ -225,4 +235,56 @@ default void waitAcls(AclBindingFilter filter, Collection<AccessControlEntry> en
         }
     }
 
+    default <B extends KafkaBroker> void verifyTopicDeletion(String topic, int numPartitions) throws Exception {
+        verifyTopicDeletion(topic, numPartitions, brokers().values());
+    }
+
+    default <B extends KafkaBroker> void verifyTopicDeletion(String topic, int numPartitions, Collection<B> brokers) throws Exception {
```

Comment:
Could you please rename it to `waitForTopicDeletion`? 

Additionally, could you please don't expose `KafkaBroker`? we will cleanup all inner interface later and that should not be used by tests anymore.